transactor/src/main/scala/io/mediachain/copycat/TransactorService.scala

```diff
@@ -129,17 +129,24 @@ extends ClientStateListener with JournalListener {
   }
   
   private def journalCommitEvent(entry: JournalEntry) {
-    if (state.streaming) {
+    if (state.streaming && entry.index >= state.index) {
+      if (entry.index != state.index) {
+        logger.warn(s"Commit skipped entries ${state.index} -> ${entry.index}")
+      }
       state.block += entry
-      state.index = entry.index
+      state.index = entry.index + 1
```

Comment:
gonna trust your math on this one, don't have a clear picture of event sequencing here